packages/vite/src/node/utils.ts

```diff
@@ -749,7 +763,7 @@ export function processSrcSet(
   replacer: (arg: ImageCandidate) => Promise<string>,
 ): Promise<string> {
   return Promise.all(
-    splitSrcSetDescriptor(srcs).map(async ({ url, descriptor }) => ({
+    parseSrcset(srcs).map(async ({ url, descriptor }) => ({
```

Comment:
This function (`processSrcSet`) is used for replacing URLs in [CSS `image-set`](https://developer.mozilla.org/en-US/docs/Web/CSS/image/image-set).
https://github.com/vitejs/vite/blob/c0edd26bbfeb9a8d80ebaa420e54fbb7f165bd9b/packages/vite/src/node/plugins/css.ts#L1748
I think we need to split the function in to two, one for `srcset` attr (the new function you wrote) and one for CSS `image-set` (you can use the original function in this PR).

Reply:
Why? Is there different behaviour? AFAIK there is not. All tests are passing. I can add some test for the usage in `image-set`. If I keep the original function for `image-set` the same bug affecting `srcset` will be present in `image-set`. This fixed it in both.

The previous version was prone to errors because of over-specification. This implementation is loose, so it doesn't know all current possible CSS functions, but is open to future extensions of possible values. So there is no future need to change it when CSS spec introduces new image functions. The whole purpose in all usages is to find potential pairs of URI and descriptor.

Reply:
Will look at it

Reply:
I reworked the Regex and the logic for `ImageCandidate` parsing. There is nothing left from the original `srcset` code, so I removed the License mention.

The Regex has named capturing groups `url` and `descriptor`. The `url` group can be any CSS function, CSS string (single or double quoted) or URL string. The `descriptor` is anything after the space and before the comma, and can have optional `type(...)` for the `image-set` case.

Reply:
Great! Thanks for handling the CSS `image-set` case as well 💚